semgrep-core/src/core/Metavariable.ml

```diff
@@ -63,13 +66,16 @@ type mvalue =
   | S of AST_generic.stmt
   | T of AST_generic.type_
   | P of AST_generic.pattern
+  | Text of string AST_generic.wrap
   (* Those can be now empty with $...XXX metavariables.
    * coupling: if you add more constructors that allow an empty content,
    * you may need to modify JSON_report.range_of_any to not get
    * some NoTokenLocation exn.
    *)
   | Ss of AST_generic.stmt list
   | Args of AST_generic.argument list
+(* This is to match the content of a string or atom, without the
```

Comment:
I guess this comment applies to `Text` but it is placed after `Args`.

Reply:
yes, I usually put the comments before ...